### PR TITLE
sql/mysql: can inspect indexes which are not support comment

### PR DESCRIPTION
**Pull request by HatsuneMiku3939**
Unfortunately, I'm using MySQL 5.1 which is not support comments on indexes.
The schema inspection for an older version of MySQL seems to be working with this PR.

I know MySQL 5.1 is too old to support. However, this PR will be helpful for unlucky people like me.

### CODE REVIEW

**Review by a8m — COMMENTED** [review]
Thanks for the contribution, @HatsuneMiku3939. See my comments and please let me know if I can help with anything. 

Just a small question - MySQL 5.5 entered its "end of life" status a long time ago. Do you mind elaborating on which platform you use it, and what version do you use? Also, any reason for not upgrading? You don't need to share this info if you can't or don't want to, I just wondered, and if you prefer to talk on Discord I'm available there too 😎

**Comment by a8m** [comment]
Hey @HatsuneMiku3939! Any update on this PR?

**Comment by HatsuneMiku3939** [comment]
> Hey @HatsuneMiku3939! Any update on this PR?

Sorry for the late reply. 
I will update this PR this weekend.

File: sql/mysql/driver.go [hunk 1]
```diff
@@ -191,6 +191,12 @@ func (d *conn) supportsRenameColumn() bool {
 	return d.gteV(v)
 }
 
+// supportsIndexComment reports if the connected database
+// supports the index comment.
+func (d *conn) supportsIndexComment() bool {
+	return d.mariadb() || d.gteV("5.5.0")
```
[the remark this answers is not in the document]
Comment (HatsuneMiku3939) [reply sql/mysql/driver.go]: @a8m 
> Index definitions now can include a comment of up to 1024 characters.

I found these sentences from release note 5.5.3.
- https://docs.oracle.com/cd/E17952_01/mysql-5.5-relnotes-en/news-5-5-3.html

[the remark this answers is not in the document]
Comment (a8m) [reply sql/mysql/driver.go]: OK, but the release note is for `5.5.3` and not `5.5.0`. BTW, this release note is from 12 years ago, any chance you can elaborate on my questions above? I wonder how many users actually use this version.

[the remark this answers is not in the document]
Comment (HatsuneMiku3939) [reply sql/mysql/driver.go]: > OK, but the release note is for 5.5.3 and not 5.5.0.
I also updated the code. The latest changes are checked against 5.5.3.

> any chance you can elaborate on my questions above?
Our company uses 5.1.x. (i don't know the patch version). And, 100+ engineers and 10+ apps are connecting this database.

Worst of all, this database is shared with other companies in our groups. We don't have control.

File: sql/mysql/inspect_test.go [hunk 2]
```diff
@@ -522,6 +524,86 @@ func TestDriver_InspectTable(t *testing.T) {
...
+					{SeqNo: 1, C: columns[3], Desc: true},
+					{SeqNo: 2, C: columns[2], Desc: true},
+				}
+				require.EqualValues(columns, t.Columns)
+				require.EqualValues(indexes, t.Indexes)
+			},
+		},
+		{
+			name:    "indexes/not_support_comment",
+			version: "5.1.60",
```
[the remark this answers is not in the document]
Comment (HatsuneMiku3939) [reply sql/mysql/inspect_test.go:568]: I tried to shrink this test case. How about this?

File: sql/mysql/driver.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (a8m) [reply sql/mysql/driver.go]: Make sense. Let me to review this again 😃

File: sql/mysql/driver.go [hunk 3]
```diff
@@ -191,6 +191,14 @@ func (d *conn) supportsRenameColumn() bool {
 	return d.gteV(v)
 }
 
+// supportsIndexComment reports if the connected database
+// supports the index comment.
+func (d *conn) supportsIndexComment() bool {
+	// According to release notes
+	// MySQL 5.5.3 starts supporting the index comment
+	return d.mariadb() || d.gteV("5.5.3")
+}
```
Comment (a8m) [inline sql/mysql/driver.go:200]:
```suggestion
// supportsIndexComment reports if the connected database
// supports comments on indexes.
func (d *conn) supportsIndexComment() bool {
	// According to Oracle release notes, comments on
	// indexes were added in version 5.5.3.
	return d.mariadb() || d.gteV("5.5.3")
}
```

File: sql/mysql/inspect.go [hunk 4]
```diff
@@ -442,6 +439,20 @@ func (i *inspect) checks(ctx context.Context, s *schema.Schema) error {
...
+	query := indexesNoCommentQuery
+	if i.supportsIndexComment() {
+		query = indexesQuery
+	}
+
+	if i.supportsIndexExpr() {
+		query = indexesExprQuery
+	}
+
+	return query
```
Comment (a8m) [inline sql/mysql/inspect.go:453]:
```suggestion
	if i.supportsIndexComment() {
		query = indexesQuery
	}
	if i.supportsIndexExpr() {
		query = indexesExprQuery
	}
	return query
```

**Review by a8m — APPROVED** [review]
💯

File: sql/mysql/driver.go [hunk 5]
```diff
@@ -191,6 +191,14 @@ func (d *conn) supportsRenameColumn() bool {
 	return d.gteV(v)
 }
 
+// supportsIndexComment reports if the connected database
+// supports comments on indexes.
+func (d *conn) supportsIndexComment() bool {
+	// According to Oracle release notes, comments on
+	// indexes were added in version 5.5.3.
+	return d.mariadb() || d.gteV("5.5.3")
+}
```
Comment (a8m) [inline sql/mysql/driver.go:200]:
```suggestion
func (c *conn) supportsIndexComment() bool {
	// According to Oracle release notes, comments on
	// indexes were added in version 5.5.3.
	return c.mariadb() || c.gteV("5.5.3")
}
```

lint fails as master version was changed with different receiver names.

**Pull request merged**